src/hotspot/cpu/x86/vm_version_x86.hpp

```diff
@@ -1044,6 +1044,21 @@ class VM_Version : public Abstract_VM_Version {
   static bool supports_clflushopt() { return ((_features & CPU_FLUSHOPT) != 0); }
   static bool supports_clwb() { return ((_features & CPU_CLWB) != 0); }
 
+  // Old CPUs perform lea on AGU which causes additional latency transfering the
+  // value from/to ALU for other operations
+  static bool supports_fast_2op_lea() {
+    return (is_intel() && supports_avx()) || // Sandy Bridge and above
+           (is_amd()   && supports_avx());   // Jaguar and Bulldozer and above
+  }
+
+  // Pre Icelake Intels have inefficient 3-op lea with 3 latency, this can be
+  // replaced by add-add or lea-add
+  static bool supports_fast_3op_lea() {
+    return supports_fast_2op_lea() &&
+           ((is_intel() && supports_clwb()) || // Icelake and above
```

Comment:
Why we use `supports_clwb ()` here?

Reply:
Icelakes introduce `clwb` instruction so I use that to filter Intel cpus.

Reply:
> Icelakes introduce `clwb` instruction so I use that to filter Intel cpus.

I'm not sure if we can find a better way (e.g., cpuid) to filter Intel CPUs.
However, this seems fine since gcc and llvm will generate `lea` even without `clwb`.